### ISSUE TITLE
Fragmenter

### ISSUE THREAD

**Issue by hesam-a** [pren/poltype#44]
Seems that fragmenter swaps some torsion prms in the parent "_copy.key_5" file. I noticed the torsion prms of 2 different baby jobs are swapped (still you can see the real values in the baby directories but in the final key_5 they are different.)

Also wish fragmenter could generate the "parentindextofragindex.txt" in all directories even in the ones that it finds the matched torsions and doesn't get into them. 

For some reason fragmenter zeros some parameters while transferring them from the fragments to the final key_5.

**Comment by walkerbdev** [comment]
Hey Hesam,

Please provide files and exact description of the problem down to what
lines are swapped and not swapped

- Brandon

**Comment by hesam-a** [comment]
Hey Brandon,

So I have run two sets of calculations and in one I ran the baby jobs separately and the other one ran it all by itself, in the latter one I got this:
torsion   401 413 418 427    -0.723 0.0 1     0.495 180.0 2    1.288 0.0 3
torsion   427 418 413 429     2.922 0.0 1    -1.748 180.0 2    0.820 0.0 3
torsion   410   424   425   427     0.0000 0.0 1     0.0000 180.0 2     0.0000 0.0 3
torsion   430   424   425   427     0.0000 0.0 1     0.0000 180.0 2     0.0000 0.0 3

however in what I did, it came out like this:
torsion   401   413   418   427    0.0000 0.0 1     0.0000 180.0 2     0.0000 0.0 3
torsion   427   418   413   429    0.0000 0.0 1     0.0000 180.0 2     0.0000 0.0 3
torsion   410   424   425   427    -0.267  0.0 1     0.239  180.0 2     0.061  0.0 3
torsion   427   425   424   430     0.877  0.0 1    -0.730  180.0 2     0.336  0.0 3

now here's the correct form which I have found by matching the indices and mapping:
torsion   401   413   418   427    -0.267  0.0 1     0.239  180.0 2     0.061  0.0 3
torsion   427   418   413   429     0.877  0.0 1    -0.730  180.0 2     0.336  0.0 3
torsion   410   424   425   427    -0.723  0.0 1     0.495  180.0 2     1.288  0.0 3
torsion   427   425   424   430     2.922  0.0 1    -1.748  180.0 2     0.820  0.0 3

now if you look at them you will see in the first one, the first 2 lines belong to the line 3 and 4 and zeros (line 3 and 4) belong to the first 2 lines, and zeros are not actually zeros it just couldn't transfer them correctly to the "_copy.key_5". Now in the second one again lines 3 and 4 belong to line 1 and 2, finally looking at the 3rd one which is the correct one you will see how they are swapped. 

I will check them all one by one to see if all parameters are correct (and in the right place) and will let you know.

Best,
Hesam

**Comment by walkerbdev** [comment]
hey Hesam,

do you have files I can look at to reproduce this easily? Specifying the paths to each would be very helpful as well (which folders)

**Comment by hesam-a** [comment]
Hey Brandon,
There are two different jobs one is yours that I've continued it and one is what I ran. I have uploaded them to the shared directory that Chengwen and I have, I'm sending you its link to your email.
Best,Hesam    On Friday, September 18, 2020, 01:33:07 PM EDT, Brandon Walker <notifications@github.com> wrote:  
 
 


hey Hesam,

do you have files I can look at to reproduce this easily? Specifying the paths to each would be very helpful as well (which folders)

**Comment by walkerbdev** [comment]
Thanks, I am not really understanding what you are trying to say here, but I will check manually now.
"now if you look at them you will see in the first one, the first 2 lines belong to the line 3 and 4 and zeros (line 3 and 4) belong to the first 2 lines, and zeros are not actually zeros it just couldn't transfer them correctly to the "_copy.key_5". Now in the second one again lines 3 and 4 belong to line 1 and 2, finally looking at the 3rd one which is the correct one you will see how they are swapped."

**Comment by hesam-a** [comment]
So there I have run two sets of calculation one is a regular and the other one I ran the baby jobs separately so poltype can just transfer the values to *copy.key_5. Now for some reason fragmenter swapped the values of line 1 and 2 with line 3 and 4 which is weird. It did the same for both jobs. I realized it when I looked at each baby job separately and noticed the values of line 1 and 2 are for line 3 and 4 and values of line 3 and 4 are for line 1 and 2.    On Friday, September 18, 2020, 02:05:25 PM EDT, Brandon Walker <notifications@github.com> wrote:  
 
 


Thanks, I am not really understanding what you are trying to say here, but I will check manually now.
"now if you look at them you will see in the first one, the first 2 lines belong to the line 3 and 4 and zeros (line 3 and 4) belong to the first 2 lines, and zeros are not actually zeros it just couldn't transfer them correctly to the "_copy.key_5". Now in the second one again lines 3 and 4 belong to line 1 and 2, finally looking at the 3rd one which is the correct one you will see how they are swapped."
